cmd/uid_provider/api/controllers/base.go

```diff
@@ -0,0 +1,34 @@
+package controller
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+
+	"gitlab.inspr.dev/inspr/core/cmd/uid_provider/client"
+)
+
+// Server is a struct that contains the variables necessary
+// to handle the necessary routes of the rest API
+type Server struct {
+	Mux *http.ServeMux
+	Rdb client.RedisManager
+}
+
+// NewServer returns a new server object
+func NewServer() Server {
+	return Server{}
+}
+
+// Init - configures the server
+func (s *Server) Init(rdb client.RedisManager) {
+	s.Mux = http.NewServeMux()
+	s.Rdb = rdb
+	s.initRoutes()
+}
```

Comment:
Combine these two functions, so that it's not reduntant

Reply:
Now that you pointed out, "NewServer( )" isn't even being used. As the Server struct is public, I'm just gonne remove the function